pkg/util/configspec.go

```diff
@@ -143,6 +145,10 @@ func SanitizeVMClassConfigSpec(
 	// Empty VmProfiles as storage profiles are disk specific
 	configSpec.VmProfile = []vimtypes.BaseVirtualMachineProfileSpec{}
 
+	configSpec.ExtraConfig = OptionValues(configSpec.ExtraConfig).Delete(
```

Comment:
more like cautionary removal from the class configSpec here?. In case the admin decides to add it?

Reply:
Yes. It is deprecated and we'll always remove it but potentially saves a step.